rust/src/encodings/binary.rs

```diff
@@ -128,6 +132,19 @@ impl<'a, T: ByteArrayType> Decoder for BinaryDecoder<'a, T> {
     async fn decode(&self) -> Result<ArrayRef> {
         self.get(..).await
     }
+
+    async fn take(&self, indices: &UInt32Array) -> Result<ArrayRef> {
+        if indices.is_empty() {
+            return Ok(new_empty_array(&T::DATA_TYPE));
+        }
+
+        // TODO: optimize for sparse index
```

Comment:
Yes, also , in case where the indices is super sparse, like `[1, 1000, 5000]`, we dont need to actually read all data from `[1:5000]`, but just issue 3 smaller I/Os.